torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -79,7 +79,23 @@ def __init__(self):
 @triton_kernel_wrapper_mutation.py_impl(DispatchKey.CompositeExplicitAutograd)
 def triton_kernel_wrapper_mutation_dense(*, kernel_idx, grid, kwargs):
     kernel = kernel_side_table.get_kernel(kernel_idx)
-    kernel[grid](**kwargs)
+    assert len(grid) != 0
+    if len(grid) == 1:
+        grid_fn = grid[0]
+    else:
+
+        def grid_fn(meta):
+            configs = kernel.configs
+            assert len(grid) == len(configs)
+            for i, (grid_val, config) in enumerate(zip(grid, configs)):
+                guards = [
+                    f"meta['{name}'] == {val}" for name, val in config.kwargs.items()
+                ]
+                guards = " and ".join(guards)
+                if eval(guards):
+                    return grid[i]
```

Comment:
Configs aren't actually guaranteed to be unique, since they can differ in num_stages/num_warps (but not kwargs).  This might be fine though, since I think the grids will be the same for duplicates.

Calling `eval()` is very expensive in python, and this function is O(num-configs).

This this could be implemented more efficiently using a dict.

Reply:
> This might be fine though, since I think the grids will be the same for duplicates

grid computation doesn't use num_warps and num_stages so duplicates are fine, just more branches.